### ISSUE TITLE
Importing scalafx.Includes._ breaks cellFactory creation

### ISSUE THREAD

**Issue by OndrejSpanel** [scalafx/scalafx#236]
Following code does not compile:

```
import scalafx.application.JFXApp
import scalafx.application.JFXApp.PrimaryStage
import scalafx.collections.ObservableBuffer
import scalafx.scene.Scene
import scalafx.scene.control.TableColumn._
import scalafx.scene.control.{TableColumn, TableView}
import scalafx.beans.property.StringProperty
import scalafx.scene.control.cell.TextFieldTableCell
import scalafx.Includes._ // does not compile
//import scalafx.util.UtilIncludes.function12jfxCallback // does not compile
//import scalafx.Includes.{function12jfxCallback => _, _} // compiles OK


class Person(name_ : String) {

  val name = new StringProperty(this, "firstName", name_)
}
object SimpleTableView extends JFXApp {

  val characters = ObservableBuffer[Person](
    new Person("Peggy"),
    new Person("Rocky")
  )

  stage = new PrimaryStage {
    title = "Simple Table View"
    scene = new Scene {
      content = new TableView[Person](characters) {
        columns ++= List(
          new TableColumn[Person, String] {
            text = "First Name"
            cellValueFactory = {_.value.name}
            cellFactory = _ => new TextFieldTableCell[Person, String]()
            prefWidth = 180
          }
        )
      }
    }
  }
}
```

The error is

>  ```Error:(34, 27) missing parameter type```
> ```cellFactory = _ => new TextFieldTableCell[Person, String]()```

As a workaround one can disable import of `function12jfxCallback `, as indicated in the source comments.

I am not sure what can be done to fix this, but the error was confusing and it was quite hard to identify the exact cause.

**Comment by jpsacha** [comment]
Looks that `function12jfxCallback` is too eager to convert lambdas with anonymous parameter to Callbacks. I think `function12jfxCallback` should be removed as too generic. Question is how many things (outside ScalaFX source code) depend on it.

**Comment by jpsacha** [comment]
To make it clear, it is possible to make the code in the initial post compile correctly be specifying lambdas argument type explicitly:
```scala
cell factory = _: TableColumn[Person, String]=> new TextFieldTableCell[Person, String]()
```
even when `import scalafx.Includes._` is used.

**Comment by jpsacha** [comment]
Removal of `function12jfxCallback` fixes the issue in Scala 2.10 and 2.11, but in 2.12.0-M4 type parameter has to be specified:
```scala
cellFactory = (_: TableColumn[Person, String]) => new TextFieldTableCell[Person, String]()
```
